_assignments/mp1.md

```diff
@@ -0,0 +1,395 @@
+
+Add a script to your trigger object by clicking `Add Component` → `New Script`, and name it `BallDropScript`. Open the script, and create the following lines:
+
+```
+public GameObject ball;
+void OnTriggerEnter(Collider other) {}
+```
+<!--
+![Ball Drop Script](/img/assignments/mp1/image11.png)
+-->
+
+The `OnTriggerEnter` function will be called when the collider attached to our empty GameObject is entered. The `other` parameter is the collider that intersected this collider.
+The `public GameObject` tag shows a neat feature of Unity.
+Save your script, then navigate to your empty trigger GameObject. In the inspector, the script component should look like this:
+
+![Script in Inspector](/img/assignments/mp1/image8.png)
+
+So our public GameObject is now a field for the script component in Unity. Drag the sphere into this field. Now, whenever you reference the `ball` variable in your script, it will be referencing the sphere you dragged in. Pretty neat! You can read more about this in the [Variables and the Inspector tutorial](https://docs.unity3d.com/Manual/VariablesAndTheInspector.html).
+
+Now, all that’s left is to make that sphere fall. You’ll need to get the rigid body of the sphere (rigid bodies deal with physics, read more here), using the `ball.GetComponent<Rigidbody>()` method. After that, simply set `rigidBody.useGravity` to `true`.
+
+### Break Out
+
+> Set up a button on the controllers to switch the player's position between an external viewing point and the room.
+
+Keyboards do not work well in VR, since the VR environment means that you cannot see the keyboard, and all of the keys largely feel the same. Controllers, with their contours and designated button shapes, are much easier to use blind. Using a controller in Unity is somewhat similar to the keyboard. Refer to the following pages for more information:
+
+- [Unity Input](https://docs.unity3d.com/Manual/ConventionalGameInput.html)
+- [Unity wiki page on Xbox controllers](http://wiki.unity3d.com/index.php?title=Xbox360Controller)
+- [OVR Input](https://developer.oculus.com/documentation/unity/latest/concepts/unity-ovrinput) (for Touch controllers)
+
+To move the player around the room, set the player’s `transform.position` to a `Vector3` corresponding to an empty spot in the room.
+Remember that the coordinates for the player refer to its center, and the player is 2 units tall. Thus, the player controller needs to be 1 unit above the ground.
+
+Make a script that switches the player between a room and a new external viewing point, e.g. a small new plane a moderate distance from the room. Pressing the button for the first time must move the player to the external viewing point. After that, pressing the button must alternate between the two locations.

```

Comment:
This part is new, correct? Why bring the xbox controls in here, rather than have them map a button to cycle the light color?

Reply:
Oversight. I think we also used keyboard in MP2, so I'll need to make sure that we tell students to set up controller input for that as well (and tell us how in the README)

Reply:
There is now a special aside detailing why we prefer controller input